### PR TITLE
[GEOT-5211] App-schema REST: data access containing non-feature types is not removed

### PR DESCRIPTION
**Pull request by ridethepenguin**
This PR implements an algorithm to automatically dispose of _hidden_ (i.e. non-accessible and non-advertised) DataAccesses when they are no longer needed.

The algorithm cannot work reliably if a polymorphic feature chaining configuration is found, so it is disabled in that case.

### CODE REVIEW

**Comment by jodygarnett** [comment]
Pull request looks fine, if this is a bug fix do you want to merge before the RC?

Or if it is a new feature please wait until next week.

**Comment by ridethepenguin** [comment]
I'd consider it a bug-fix, also Ben called the issue "a resource leak worth fixing".

@bencaradocdavies For me, It'd be great to have this merged before the RC, what is your opinion?

**Comment by bencaradocdavies** [comment]
@ridethepenguin I am running a full online test at this very moment. I agree that this is a bug in app-schema+REST so qualifies for inclusion in the RC. I hope to complete my review of this PR quite soon.

File: modules/extension/app-schema/app-schema/src/main/java/org/geotools/data/complex/AppSchemaDataAccess.java [hunk 1]
```diff
@@ -93,6 +93,14 @@
     private FilterFactory2 filterFac = CommonFactoryFinder.getFilterFactory2(null);
 
     /**
+     * Flag to mark non-accessible data accesses, which should be automatically
+     * disposed of when no longer needed by any accessible data access.
+     */
+    boolean hidden = false;
+
+    List<Name> relatedTypeNames = new ArrayList<>();
```
Comment (bencaradocdavies) [inline modules/extension/app-schema/app-schema/src/main/java/org/geotools/data/complex/AppSchemaDataAccess.java]: @ridethepenguin this new member relatedTypeNames is not used anywhere and should be removed.

**Comment by bencaradocdavies** [comment]
That looks great, @ridethepenguin. Your unit test coverage is solid, and full GT + GWC + GS + GS app-schema online tests against postgis all passed. I found one unused member that should be removed (see above), then please go ahead and merge on master.

File: modules/extension/app-schema/app-schema/src/main/java/org/geotools/data/complex/AppSchemaDataAccess.java (hunk 1 above)
Comment (ridethepenguin) [inline modules/extension/app-schema/app-schema/src/main/java/org/geotools/data/complex/AppSchemaDataAccess.java]: @bencaradocdavies Indeed, it's a remnant from a discarded design, I'll
remove it tomorrow

Should I add a commit to the PR or rebase? I remember you advising against
rebasing a PR branch...
Il 10/set/2015 23:14, "Ben Caradoc-Davies" <notifications@github.com> ha
scritto:

**Comment by bencaradocdavies** [comment]
@ridethepenguin @jodygarnett I pushed the removal of the unused member in a separate commit on master. All done!

**Pull request merged**